### ISSUE TITLE
Profile with/out attn_pdrop in gpt2, send to jax devs

### ISSUE THREAD

**Issue by dlwh** [stanford-crfm/levanter#210]
I observe a massive speed hit when using dropout on TPU:

gpt2-small throughput (v3-32)

* no attention dropout: 0.3654
* threefry_partitionable dropout: 0.5499
* with unsafe_rbg, no tpu initflag: 3.92 (!!!)

Jax devs have asked for a repro and/or profile.

- [ ] profile
- [ ] repro (ideally)

**Comment by Ivan-Zhou** [comment]
Here's the configuration that I use for profiling:
- [gpt2_small_fast.yaml](https://github.com/stanford-crfm/levanter/blob/jax-profile/config/gpt2_small_fast.yaml), including changes of:
  - attn_pdrop=0.1
  - use_biases=False
- FLAG=unsafe_rbg ([here](https://github.com/stanford-crfm/levanter/blob/6ad785de086e491ff27da46076d92849b7d0df7d/src/levanter/main/train_lm.py#L34))

**Comment by Ivan-Zhou** [comment]
What is `repro`? Giving them a branch to reproduce the result?

**Comment by dlwh** [comment]
well ideally it would be a minimized repro, which is kind of a bunch of work. let's see if they'll settle for the profile? (They indicated they might?)

**Comment by Ivan-Zhou** [comment]
Maybe I misunderstood, but the impact of `unsafe_rbg` seems to be against with the hypothesis. I observed that, after enabling `unsafe_rbg`, the training throughput gets a massive hit and the maximum batch size I could set also decreased to below 4096. I thought `unsafe_rbg` is supposed to help "reduce the size of the Tensorflow graph and compilation time".

I captured the observation here: https://github.com/stanford-crfm/levanter/issues/226

**Comment by Ivan-Zhou** [comment]
I do observe the hit on training throughput when tuning attn_pdrop from 0.1 to 0:
![image](https://github.com/stanford-crfm/levanter/assets/8097756/824ef30c-1214-41e8-aecd-92f1bd41351c)

**Comment by Ivan-Zhou** [comment]
(the earlier comment was stated wrongly and has been removed)
The results are aligned with the observation stated at the beginning of this issue: after adding attention dropout, the training throughput decreases from 770K tokens per second to 500K.

**Comment by Ivan-Zhou** [comment]
I made a writeup with my experiment results and profiling data: https://docs.google.com/document/d/1C6M8IJmOFFtG_KmAv2yxkVmSuryFbPUmfFccth1LUo4/edit
@dlwh If this looks good to you, we can share it with Jax people.

**Comment by dlwh** [comment]
lgtm, yeah, let's send it over. Hopefully they can figure it out without us having to minimize
